pkg/lib/cockpit-components-table.scss

```diff
@@ -1,14 +1,90 @@
 @import "~@patternfly/patternfly/components/Table/table.scss";
 @import "~@patternfly/patternfly/components/Table/table-grid.scss";
 
-// Display the actions next to the header inline
-header {
-    display: flex;
-    > :only-child {
+.ct-table {
+    &.pf-m-compact {
+        > thead, > tbody {
+            > tr:not(.pf-c-table__expandable-row) {
+                // We actually want the normal font size for our lists
+                --pf-c-table-cell--FontSize: var(--pf-global--FontSize--md);
+            }
+        }
+    }
+
+    &-header {
+        display: flex;
+        align-items: center;
+        flex-wrap: wrap;
+
+        > :only-child {
+            flex: auto;
+        }
+    }
+
+    &-heading {
+        // Push buttons to the right by stretching the heading
         flex: auto;
+        // Add a bit of minimum margin to the right of the heading
+        margin-right: var(--pf-global--spacer--md);
+        // Set a minimum height of 3rem, so when buttons wrap, there's spacing
+        min-height: var(--pf-global--spacer--2xl);
+        // Make sure textual content is aligned to the center
+        display: flex;
+        align-items: center;
     }
-}
 
-thead.listing-ct-empty tr td {
-    font-weight: normal;
-}
+    &-actions {
+        > * {
+            margin: var(--pf-global--spacer--xs) 0 var(--pf-global--spacer--xs) var(--pf-global--spacer--sm);
+        }
+
+        > :first-child {
+            margin-left: 0;
+        }
+    }
+
+    &-empty {
+        tr td {
+            // FIXME: Not sure if this is needed. Was a direct port.
+            font-weight: normal;
+        }
+    }
+
+    // Fix toggle button alignment
```

Comment:
The toggle button upstream behaves correctly. Can you confirm or reject?

Reply:
But upstream alignment actually _doesn't_ work when there are widgets in a row, especially buttons. Hence the need for this.